### ISSUE TITLE
Would be nice to have `Running()` and `Terminating()`

### ISSUE THREAD

**Issue by arvenil** [go-tomb/tomb#19]
Consider below pseudo code

```go
type Job struct {
    tomb *tomb.Tomb
}

func (j *Job) Start() error {
    if j.tomb.Alive() {
        return errors.New("job is still running")
    }
    j.tomb = &tomb.Tomb{}
    j.tomb.Go()
}
```

I want to create service that has long running goroutine, but only one, above code is not safe because `Alive()` can return `false` when tomb is in Dying state. So, I would like to know if tomb is Running (so dying or alive) or not. And there doesn't seem to be a way to actually do this with tomb.

I could read `Dead()` chan, but this would work only when tomb died, but if it never was started then it would hang forever as it is initialized as open chan.

Easy way would be something like this:
```
func (t *Tomb) Running() bool {
	t.m.Lock()
	defer t.m.Unlock()
	return t.alive > 0
}
```

`Terminating()` would be also useful since we don't always wait on channels in goroutines, but this can be easily done even without modifications to tomb.

```
func (t *Tomb) Terminating() bool {
        select {
            case <-t.dying:
                return true
            default:
                return false 
        }
}
```
